FrameworkSystem/private/standardLogging/gLogging.py

```diff
@@ -0,0 +1,335 @@
+  set and get the level of the embedded logger and create log messages with it.
+
+  gLogging could delegate the initialization and the configuration to a factory of the root logger be it can not
+  because it has to wrap the old gLogger.  
+  """
+
+  _componentName = "Framework"
+
+  def __init__(self, father=None, fathername='', name=''):
+    """
+    Initialization of the gLogging object.
+    :params fathername: string representing the name of the father logger in the chain.
+    :params name: string representing the name of the logger in the chain. 
+    :params father: gLogging father of this new gLogging.
+    By default, 'fathername' and 'name' are empty, because getChild accepts only string and the first empty
+    string corresponds to the root logger. 
+
+    Example: 
+    logging.getLogger('') == logging.getLogger('root') root logger
+    logging.getLogger('root').getChild('log') == root.log == log child of root
+    """
+
+    # gLogging chain
+    self._childrens = []
+    self._parent = father
+
+    # all the different backends
+    self._backendsDict = {'stdout': StdoutBackend(),
+                          'stderr': StderrBackend(),
+                          'file': FileBackend(),
+                          'server': RemoteBackend()}
+
+    # initialize display options with the ones of the gLogging parent
+    if self._parent is not None:
+      self._options = self._parent.getDisplayOptions()
+    else:
+      self._options = {'showHeaders': True, 'showThreads': False, 'Color': False, 'Path': False}
+
+    # dictionary of the option state, modified by the user or not
+    # this is to give to the options the same behaviour that the logging setLevel()
+    self._optionsmodified = {'showHeaders': False, 'showThreads': False}
+
+    self._backendsList = []
+
+    self._logger = logging.getLogger(fathername).getChild(name)
+
+  def showHeaders(self, yesno=True):
+    """
+    Depending on the value, display or not the prefix of the message.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showHeaders', yesno)
+
+  def showThreadIDs(self, yesno=True):
+    """
+    Depending on the value, display or not the thread ID.
+    :params yesno: boolean determining the behaviour of the display
+    """
+    self._setOption('showThreads', yesno)
+
+  def _setOption(self, optionName, value):
+    """
+    Depending on the value, modify the value of the option.
+    This option will not be modified anymore. 
+    The options of the childrens will be updated if they were not modified before by a developer
+    :params optionName: string representing the name of the option to modify
+    :params value: boolean to give to the option  
+    """
+    self._options[optionName] = value
+    self._optionsmodified[optionName] = True
+    self._setChildrensDisplayOptions(optionName, self._options)
+    self._updateFormat()
```

Comment:
updateFormat() ? I added a little comment 
It is to create a new formatter object for the handler in the backend object, in order to modify the display of the log messages when an option changes. Do you understand ?